### PR TITLE
llvm: "master" branch is now "main" branch

### PR DESCRIPTION
**Pull request by eugeneswalker**
llvm-project no longer has `master` branch -- it is now called `main` branch

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]
@trws @naromero77

**Comment by mwkrentel** [comment]
@eugeneswalker 
Could you run `spack info llvm` with this change?
Where does `main` appear in the order?

I don't think `main` is a special version name, I think it's only
master, develop, trunk, and head in some order.

The problem is that you may be putting main in the wrong order which
could change some decisions on master vs. main.

**Comment by adamjstewart** [comment]
> I don't think main is a special version name, I think it's only master, develop, trunk, and head in some order.

See `lib/spack/spack/version.py`, it's:
```python
# Infinity-like versions. The order in the list implies the comparison rules    
infinity_versions = ['develop', 'main', 'master', 'head', 'trunk']  
```
so `main` is fine.

**Comment by adamjstewart** [comment]
@eugeneswalker let's hold off on the auto-merge until the maintainers approve.

**Comment by eugeneswalker** [comment]
> @eugeneswalker
> Could you run `spack info llvm` with this change?
> Where does `main` appear in the order?
> 
> I don't think `main` is a special version name, I think it's only
> master, develop, trunk, and head in some order.
> 
> The problem is that you may be putting main in the wrong order which
> could change some decisions on master vs. main.

```
(base) ➜  spack git:(llvm-master-is-now-main) spack info llvm
CMakePackage:   llvm

...
Preferred version:
    11.0.1    https://github.com/llvm/llvm-project/archive/llvmorg-11.0.1.tar.gz

Safe versions:
    main      [git] https://github.com/llvm/llvm-project on branch main
    11.0.1    https://github.com/llvm/llvm-project/archive/llvmorg-11.0.1.tar.gz
    11.0.0    https://github.com/llvm/llvm-project/archive/llvmorg-11.0.0.tar.gz
    10.0.1    https://github.com/llvm/llvm-project/archive/llvmorg-10.0.1.tar.gz
    10.0.0    https://github.com/llvm/llvm-project/archive/llvmorg-10.0.0.tar.gz
    9.0.1     https://github.com/llvm/llvm-project/archive/llvmorg-9.0.1.tar.gz
...
```

**Comment by mwkrentel** [comment]
> Infinity-like versions. The order in the list implies the comparison rules    
> infinity_versions = ['develop', 'main', 'master', 'head', 'trunk']  

Ok, I was forgetting `main`, never mind, not a problem.

Btw, I wish `stable` were an infinity name.  There are lots of
projects that have develop and stable branches.

**Comment by adamjstewart** [comment]
> Btw, I wish stable were an infinity name. There are lots of projects that have develop and stable branches.

Submit a PR?

**Comment by mwkrentel** [comment]
> Submit a PR?

That's more of a policy issue rather than technical.
Do you think there would be support for this?

Btw, I see where 'main' was added in Aug 2020.
So my memory isn't quite as bad as I had thought. :-)

**Comment by adamjstewart** [comment]
I'm not opposed to it. Personally I think all non-numeric versions should sort higher, but I tried that and it was rejected.

**Review by trws — APPROVED** [review]
Works for me.  FWIW having stable in the list seems reasonable.

**Comment by sethrj** [comment]
Version ordering in spack is an unsolved problem.

**Comment by naromero77** [comment]
LGTM

**Pull request merged**
